tests/test_comments.py

```diff
@@ -0,0 +1,72 @@
+import pytest
+import pygohcl
+
+
+@pytest.mark.parametrize(
+    "hcl,expected",
+    [
+        (  # Simple inline comments
+            """variable "test" {
+                validation {
+                    condition = alltrue(
+                        # This is a comment on it's own line
+                        contains(["a", "b"], var) // This is a trailing comment
+                    )
+                }
+            }""",
+            {"variable": {"test": {"validation": {"condition": "alltrue( contains([\"a\", \"b\"], var) )"}}}},
+        ),
+        (  # Inline comment inside of a string literal
+            """variable "test" {
+                validation {
+                    condition = contains(["a", "b #not a comment"], var) # This is a comment
+                }
+            }""",
+            {"variable": {"test": {"validation": {"condition": "contains([\"a\", \"b #not a comment\"], var)"}}}},
+        ),
+        (  # Block comment
+            """variable "test" {
+                validation {
+                    condition = alltrue(
+                        /* This is a
+                            block comment */
+                        contains(["a", "b"], var)
+                    )
+                }
+            }""",
+            {"variable": {"test": {"validation": {"condition": "alltrue( contains([\"a\", \"b\"], var) )"}}}},
+        ),
+        (  # Block comment inside of a string literal
+            """variable "test" {
+                validation {
+                    condition = contains(["a", "b /*not a comment*/"], var) /* This is a comment */
+                }
+            }""",
+            {"variable": {"test": {"validation": {"condition": "contains([\"a\", \"b /*not a comment*/\"], var)"}}}},
+        ),
+        (  # Mixed comments, nested string literals
+            """variable "test" {
+                validation {
+                    condition = alltrue([ # Who
+                        for val in var.values: // would
+                            /* comment
+                               like */
```

Comment:
@petroprotsakh please add something like this, when the inline comment is inside a block comment on the same line as a terminal sequence for the block comment `*/`
```suggestion
                             //  like */
```

Reply:
and any other similar combinations

Reply:
@recipe Good catch.
Improved the parser and added more tests.
I'll merge the PR and prepare the release if you're happy with it.

Reply:
@petroprotsakh yes, go ahead.